### ISSUE TITLE
 createDriver() method throws an java.lang.NullPointerException  

### ISSUE THREAD

**Issue by imikulich** [qaprosoft/carina#183]
Logs:

```
2017-11-22` 11:53:29] [AuthService.java] [23] [INFO] Check user status for registration: mfpauto61
[2017-11-22 11:53:29] [AuthService.java] [23] [INFO] identified free user: mfpauto61
[2017-11-22 11:54:59] [DriverPool.java] [23] [ERROR] Driver initialization 'default' FAILED for selenium: null! Retry 1 of 1 time - Unable to create new remote session. desired capabilities = Capabilities [{app=/opt/apk/myfitnesspal-develop-10593-qa-release.apk, noSign=true, newCommandTimeout=180, STF_ENABLED=true, autoGrantPermissions=true, platformName=Android, deviceName=ANY}], required capabilities = Capabilities [{}]
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'e8ac6c32c60b', ip: '172.18.0.43', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.0-98-generic', java.version: '1.8.0_131'
Driver info: driver.version: AndroidDriver
[2017-11-22 11:55:09] [BaseMobileTest.java] [23] [ERROR] Attempt #0 failed:org.openqa.selenium.SessionNotCreatedException: Unable to create new remote session. desired capabilities = Capabilities [{app=/opt/apk/myfitnesspal-develop-10593-qa-release.apk, noSign=true, newCommandTimeout=180, STF_ENABLED=true, autoGrantPermissions=true, platformName=Android, deviceName=ANY}], required capabilities = Capabilities [{}]
Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
System info: host: 'e8ac6c32c60b', ip: '172.18.0.43', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.0-98-generic', java.version: '1.8.0_131'
Driver info: driver.version: AndroidDriver
[2017-11-22 11:55:37] [Device.java] [23] [INFO] adb connect ua.qaprosoft.com:7752
[2017-11-22 11:55:38] [DevicePool.java] [23] [INFO] register device fot current thread id: 23; device: 'Samsung_Galaxy_S8_Plus'
[2017-11-22 11:55:39] [DriverPool.java] [23] [ERROR] Driver initialization 'default' FAILED for selenium: null! Retry 1 of 1 time - null
[2017-11-22 11:55:49] [TestNamingUtil.java] [23] [WARN] Unable to find registered test name for threadId: 23. stack is null!
[2017-11-22 11:55:49] [TestNamingUtil.java] [23] [WARN] Unable to find registered test name for threadId: 23. stack is null!
[2017-11-22 11:55:49] [Messager.java] [23] [INFO] 
======================================================================================================================================
INFO: TEST [C16500 Favorites Lists - createRecipe] STARTED at [23:55:49 2017-11-22]
[2017-11-22 11:55:49] [ZafiraConfigurator.java] [23] [WARN] Unable to detect current device for threadId: 23
[2017-11-22 11:55:49] [Messager.java] [23] [INFO] 
INFO: TEST [C16500 Favorites Lists - createRecipe] SKIPPED at [23:55:49 2017-11-22] - java.lang.NullPointerException

com.qaprosoft.carina.core.foundation.webdriver.DriverPool.createDriver(DriverPool.java:317)
com.qaprosoft.carina.core.foundation.webdriver.DriverPool.restartDriver(DriverPool.java:188)
com.qaprosoft.carina.core.foundation.webdriver.DriverPool.restartDriver(DriverPool.java:152)
com.myfitnesspal.android.autotest.test.BaseMobileTest.login(BaseMobileTest.java:107)
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:84)
org.testng.internal.Invoker.invokeConfigurationMethod(Invoker.java:564)
org.testng.internal.Invoker.invokeConfigurations(Invoker.java:213)
org.testng.internal.Invoker.invokeConfigurations(Invoker.java:138)
org.testng.internal.TestMethodWorker.invokeBeforeClassMethods(TestMethodWorker.java:175)
org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:107)
org.testng.TestRunner.privateRun(TestRunner.java:767)
org.testng.TestRunner.run(TestRunner.java:617)
org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
org.testng.SuiteRunner.access$000(SuiteRunner.java:38)
org.testng.SuiteRunner$SuiteWorker.run(SuiteRunner.java:382)
org.testng.internal.thread.ThreadUtil$2.call(ThreadUtil.java:64)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
java.lang.Thread.run(Thread.java:748)
```

**Comment by vdelendik** [comment]
good catch! one of the latest changes in Device constructor is a reason

**Comment by vdelendik** [comment]
should be fixed in 5.2.1.24
https://github.com/qaprosoft/carina/pull/186
